### PR TITLE
Add optional cancellation tokens again in `DurableTaskClient`

### PR DESCRIPTION
**Pull request by davidmrdavid**
Between tags 1.1.1. and 1.2.0 (https://github.com/microsoft/durabletask-dotnet/compare/v1.1.1...v1.2.0#diff-c7a6ad429a79519f6c23e141c7d033c7066d66ef34af21302c877fbdd62171c7R373), we made several changes to `DurableTaskClient` that unfortunately changed regressed some behavior.

In particular, we made the following 3 changes:

| Before | After |
| -------| ------|
|     public virtual Task TerminateInstanceAsync(string instanceId, object? output = null, CancellationToken cancellation = default) |     public virtual Task TerminateInstanceAsync(string instanceId, object? output, CancellationToken cancellation = default)|
|     public virtual Task<PurgeResult> PurgeInstanceAsync(string instanceId, CancellationToken cancellation = default) |     public virtual Task<PurgeResult> PurgeInstanceAsync(string instanceId, CancellationToken cancellation) |
|     public virtual Task<PurgeResult> PurgeAllInstancesAsync(PurgeInstancesFilter filter, CancellationToken cancellation = default) |      public virtual Task<PurgeResult> PurgeAllInstancesAsync(PurgeInstancesFilter filter, CancellationToken cancellation) |

In other words, we made several previously "optional" parameters mandatory. This caused .NET to start dispatching **different** implementations for this method, which led to users seeing "method not supported" exceptions, as reported here: https://github.com/microsoft/durabletask-dotnet/issues/282#issuecomment-2114590648

This PR aims to bring back the original interfaces by making those parameters optional again, in hopes to restore the original method dispatch resolution.

_Still to be tested._

### CODE REVIEW

**Comment by davidmrdavid** [comment]
Converting this to draft due to ambiguity in the resolution still

**Comment by davidmrdavid** [comment]
Abandoning, I think this can be resolved fully by implementing the interfaces in the worker extension.
